core/src/main/java/org/mule/runtime/core/internal/message/InternalEvent.java

```diff
@@ -45,6 +45,21 @@
   @Deprecated
   <T> T getInternalParameter(String key);
 
+  /**
+   * @return a {@link EventInternalContext} with state from source-flow dispatch
+   * @since 4.4, 4.3.1
+   */
+  public EventInternalContext getFlowProcessMediatorPhaseContext();
+
+  /**
+   * Sets context related to the source-flow dispatch
+   * 
+   * @return a {@link EventInternalContext} with state from source-flow dispatch
+   * @param context an {@link EventInternalContext}
+   * @since 4.4, 4.3.1
+   */
+  public void setFlowProcessMediatorPhaseContext(EventInternalContext flowProcessMediatorPhaseContext);
```

Comment:
I would remove the `Phase` thing from this name. The Phase concept is genetic garbage inherited from 3.x... I wouldn't bubble that up all the way here